python/pyspark/sql/tests.py

```diff
@@ -429,6 +429,17 @@ def test_udf_with_input_file_name(self):
         row = self.spark.read.json(filePath).select(sourceFile(input_file_name())).first()
         self.assertTrue(row[0].find("people1.json") != -1)
 
+    def test_udf_with_decorator(self):
+        from pyspark.sql.decorators import udf
+        from pyspark.sql.types import IntegerType
+
+        @udf(IntegerType())
```

Comment:
It does. In general decorators with arguments should be called with empty parens. This call is equivalent to:

    udf()(truncate)

We can support parentheses-less call but it requires a bit ugly checks on runtime. I'll push an example implementation and additional tests in a moment.